src/Components/Button.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react';
+
+function Button({ bgColor, hoverColor, textColor, children , border}) {
+    const baseClass = 'font-bold py-4 px-16 rounded-full text-lg min-w-max flex items-center gap-x-2 transition-all';
+    const [isHovered, setIsHovered] = useState(false);
+    const colorClass = `${bgColor} ${textColor} ${border} ${isHovered ? hoverColor : ''}`;
+
+    return (
+        <button 
+            className={`${baseClass} ${colorClass}`}
+            onMouseEnter={() => setIsHovered(true)}
+            onMouseLeave={() => setIsHovered(false)}
+        >
+            {React.Children.map(children, child => {
```

Comment:
haaaaackyyyyy, there are simpler approaches to it

Reply:
this is why I like to use currentColor to allow the icons to inherit the same text color. with that you can allow the button to control the color of both the text and the icons

Reply:
also checkout groups https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state